plugins/@grouparoo/cloudwatch/src/initializers/plugin.ts

```diff
@@ -0,0 +1,53 @@
+import { Initializer, loggers, env, ActionheroLogLevel } from "actionhero";
+import { URL } from "url";
+import * as winston from "winston";
+import WinstonCloudWatch from "winston-cloudwatch";
+
+const packageJSON = require("./../../package.json");
+
+export default class CloudwatchInitializer extends Initializer {
+  constructor() {
+    super();
+    this.name = packageJSON.name;
+    this.loadPriority = 1;
+  }
+
+  async initialize() {
+    if (!process.env.S3_ACCESS_KEY) return;
+    if (!process.env.S3_SECRET_ACCESS_KEY) return;
+    if (!process.env.S3_REGION) return;
```

Comment:
These are the same env vars used in other parts of the app, right? Something interesting here is that we would be pointing these to our own S3 bucket, so other things like loading csvs from s3 won't work, is that right? I think these should be distinct so that users can set up s3 with their details if necessary.

We would also be passing our s3 credentials to the app now, which depending on how we set it up might mean this access key having access to push other people's logs as well if they somehow get to the env variables. They can't be accessed through calculated properties and I believe that's the only code hosted users can inject at the moment, so it's fine. Just something to keep in mind. It might be good to give each of these apps a different key that restricts access to their stuff. More to talk about here when we set it up.

Reply:
Well, we did recently remove the files systems, so there is no other place in the App that uses `AWS*` environment variables.  In the cloud app, since you are pre-signing the config download, I *think* this is the only S3 Auth in use?

For customers deploying on their own, I *think* they would want to make only one s3 account "for the app", and then give that user only the permissions it needs.

Reply:
ah! you're right- these env variables were required when _uploading_ files to s3 which we no longer support. I think this is ok then.